### PR TITLE
[GQL-8] #4314 console: converting routes.js and server.js to typescript fix 

### LINKED ISSUE #7: console: migrate to TypeScript #4314
1. Convert serve.js and routes.js file from to .ts or .tsx.
2. Add types, fix all type errors.
3. Submit a PR! 🎉

### PR DESCRIPTION
**Pull request by sudeshna-22**
### Description

Server and routes file was initially in the JavaScript. So, it was converted into the TypeScript as per requirement. 

### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.
- [x] No changelog-required

### Affected components

- [x] Console

### Related Issues

console: migrate to TypeScript #4314 close #7

### Solution and Design

The issue solved in this PR doesnot change any design. As mentioned above the files were converted into the typescript by changing the extension from js to ts, defining the existing types and fixing the errors.

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [ ] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

File: console/src/routes.tsx [hunk 1]
```diff
@@ -61,16 +63,22 @@ const routes = store => {
...
-      cb();
+      if (typeof cb === 'function') {
+        cb();
+      }
     }
 
     return;
   };
 
+  console.log(store, 'storetype');
```
Comment (aayushdura) [inline console/src/routes.tsx:80]: Please do remove these console logs

File: console/package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "--- DEV ------------------------------------------------": "",
     "dev": "concurrently --kill-others \"npm run watch-client\" \"npm run start-dev\"",
     "start-dev": "node -r dotenv/config ./bin/server.js",
-    "watch-client": "TAILWIND_MODE=watch node -r dotenv/config webpack/webpack-dev-server.js",
+    "watch-client": "node -r dotenv/config webpack/webpack-dev-server.js",
```
Comment (sauzzon) [inline console/package.json:20]: Don't commit this file

**Review by sauzzon — COMMENTED** [review]
Update PR with description, screenshots and video ( if any) , steps to test and so on.

File: rfcs/transforms.md [hunk 3]
```diff
@@ -10,207 +10,72 @@ The schema for `transforms` is described here:
...
 data Transforms
   = Transforms
   { requestMethod ::  Maybe RequestMethod
  -- ^ Change the request method to one provided here. Nothing means POST.
-
   , requestURL    ::  Maybe TemplateText
  -- ^ Change the request URL to one provided here. Nothing means original URL.
-
-  , request_body  ::  Maybe TemplateText
+  , request_body  ::  Maybe TemplateText                  # Raw templating
```
Comment (sauzzon) [inline rfcs/transforms.md:20]: Don't commit files that you haven't changed.

**Review by sauzzon — CHANGES_REQUESTED** [review]
There are 3 commits that are done by other people. Revert those.

File: console/src/routes.tsx [hunk 4]
```diff
@@ -74,21 +80,22 @@ const routes = store => {
...
-    composeOnEnterHooks,
-    () => {
-      return {};
-    },
-    () => {
-      return { public: [], personal: [] };
-    }
+    composeOnEnterHooks
+    // () => {
+    //   return {};
```
Comment (shrestharajani) [inline console/src/routes.tsx:85]: I think comments can be removed.

**Pull request closed without merging**